General scope 2066 (#2116)
Codecademy/docs: content/general/concepts/scope/scope.md

            set loopVariable = i
            display loopVariable
            
        display blockVariable
        
    display globalVariable
    display blockVariable
    display loopVariable
end function
```
In this pseudocode, the following scopes exist:

1. Global Scope: The 'globalVariable' is declared outside any functions or blocks, making it accessible from anywhere within the program, including all functions and blocks.
2. Local Scope: The 'blockVariable' is declared within the if-statement block. It is only accessible within that block and any nested blocks, such as the for-loop. Trying to access 'blockVariable' outside of its block would result in an error. The 'loopVariable' is declared within the for-loop block. It has scope limited to that block and is only accessible during each iteration of the loop. Attempting to access 'loopVariable' outside of the loop would result in an error.

The pseudocode snippet demonstrates how variables can have different scopes depending on where they are declared. Understanding scope is essential for correctly accessing variables and avoiding conflicts between different parts of a program.

## More information

> You can review more information regarding the concept of scope for [Python](https://www.codecademy.com/resources/docs/python/scope), [Ruby](https://www.codecademy.com/resources/docs/ruby/scope), and [Javascript](https://www.codecademy.com/resources/docs/javascript/variable-scope)

Reviewer: [Nitpick] Pronouns are not allowed according to the [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#writing-entries), so I came up with another way to convey this paragraph's message. 
```suggestion
Hungry for more information? Consider going to the docs section on Codeacademy to learn about the concept of scope in [Python](https://www.codecademy.com/resources/docs/python/scope), [Ruby](https://www.codecademy.com/resources/docs/ruby/scope), and [Javascript](https://www.codecademy.com/resources/docs/javascript/variable-scope)
```

Author: Just did ! Thank you @CBID2 !

---
Title: 'Scope'
Description: 'Visibility and accessibility of variables, functions, and objects within a particular part of a program.'
Subjects: 
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Accessibility'
  - 'Variables'
  - 'Visibility'
CatalogContent:
  - 'paths/computer-science'
---

**Scope** in programming refers to the visibility and accessibility of variables, functions, and objects within a particular part of a program. It defines where these entities can be accessed and how long their values persist during program execution. Understanding scope is crucial for writing maintainable and bug-free code, as it helps organize and manage variables, prevents naming conflicts, and ensures proper memory management.

## Example

To illustrate the concept of scope, consider the following pseudocode snippet, which demonstrates various types of scope using common programming structures:

```pseudo
function main()
  set globalVariable = 10

  if true:
    set blockVariable = 20
        
    for i = 1 to 3:
      set loopVariable = i
      display loopVariable
            
    display blockVariable
        
  display globalVariable

>   display blockVariable

Reviewer 2: ```suggestion
```
You can't access the blockVariable here.

>   display loopVariable

Reviewer 2: ```suggestion
```
You can't access loopVariable here.

end function
```

In this pseudocode, the following scopes exist:

1. Global Scope: The `globalVariable` is declared outside any functions or blocks, making it accessible from anywhere within the program, including all functions and blocks.

> 2. Local Scope: The `blockVariable` is declared within the `if`-statement block. It is only accessible within that block and any nested blocks, such as the `for`-loop. Trying to access `blockVariable` outside of its block would result in an error. The `loopVariable` is declared within the `for`-loop block. It has scope limited to that block and is only accessible during each iteration of the loop. Attempting to access `loopVariable` outside of the loop would result in an error.

Reviewer 3: @rugj84 This explanation can stay
